src/include/pipeline/dsm_cqueue.h

```diff
@@ -42,24 +44,19 @@ typedef struct dsm_cqueue
 	int magic;
 
 	LWLock  lock;
-	LWLock  *ext_lock;
 
 	int        size; /* physical size of buffer */
-	dsm_handle handle;
 
-	/*
-	 * These are offsets from the address of the dsm_cqueue. Pointers don't work with
-	 * dsm because the memory segment can be mapped at different addresses.
-	 */
-	atomic_ullong  head;
-	atomic_ullong  tail;
-	atomic_ullong  cursor;
+	atomic_ullong head;
+	atomic_ullong tail;
+	atomic_ullong cursor;
 
 	atomic_uintptr_t producer_latch;
 	atomic_uintptr_t consumer_latch;
 
-	dsm_cqueue_pop_fn pop_fn;
-	dsm_cqueue_cpy_fn cpy_fn;
+	dsm_cqueue_peek_fn peek_fn;
```

Comment:
Can you add some comments here or somewhere else that explain what each of these hooks is for?